admin/assets/javascripts/discourse/models/repo.js

```diff
@@ -94,15 +98,59 @@ export default class Repo {
       }
     }
 
+    if (attributes.plugin) {
+      this.plugin = AdminPlugin.create(attributes.plugin);
+    }
+
     for (const [key, value] of Object.entries(attributes)) {
-      if (key === "latest") {
+      if (["latest", "plugin"].includes(key)) {
         continue;
       }
 
       this[key] = value;
     }
   }
 
+  @cached
+  get nameTitleized() {
+    if (this.plugin) {
+      return this.plugin.nameTitleized;
+    }
+
+    let name = this.name
+      .split(/[-_]/)
+      .map((word) => {
+        return capitalize(word);
+      })
+      .join(" ");
+
+    // Cuts down on repetition.
+    const discoursePrefix = "Discourse ";
+    if (name.startsWith(discoursePrefix)) {
+      name = name.slice(discoursePrefix.length);
+    }
+
+    return name;
+  }
+
+  get linkUrl() {
+    if (this.plugin) {
+      return this.plugin.linkUrl;
+    }
+
+    return this.url;
+  }
+
+  get author() {
+    if (this.plugin) {
+      return this.plugin.author;
+    } else if (this.name === "docker_manager") {
```

Comment:
This is unnecessary -- docker_manager is an official plugin owned by us, so it should do this by default with `this.plugin.author`:

https://github.com/discourse/discourse/blob/792e66966c3006ec4abe9746d1554a44fcbe0c2c/app/assets/javascripts/admin/addon/models/admin-plugin.js#L88-L94

Reply:
Same as https://github.com/discourse/docker_manager/pull/214#discussion_r1667699878. I needed this since the docker_manager plugin is a hidden plugin and I was fetching plugin information only for the visible plugins. It's no longer the case after the commit https://github.com/discourse/docker_manager/pull/214/commits/662fee0d3f7c390d7fad24aa665db539ac7e000a